object-handlers.go

```diff
@@ -106,24 +113,22 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 
 	// Get request range.
 	var hrange *httpRange
-	if hrange, err = parseRequestRange(r.Header.Get("Range"), objInfo.Size); err != nil {
-		// Handle only errInvalidRange
-		// Ignore other parse error and treat it as regular Get request like Amazon S3.
-		if err == errInvalidRange {
-			writeErrorResponse(w, r, ErrInvalidRange, r.URL.Path)
-			return
+	rangeHeader := r.Header.Get("Range")
+	if rangeHeader != "" {
+		if hrange, err = parseRequestRange(rangeHeader, objInfo.Size); err != nil {
+			// Handle only errInvalidRange
+			// Ignore other parse error and treat it as regular Get request like Amazon S3.
+			if err == errInvalidRange {
+				writeErrorResponse(w, r, ErrInvalidRange, r.URL.Path)
+				return
+			}
+
+			// log the error.
+			errorIf(err, "Invalid request range")
 		}
 
-		// log the error.
-		errorIf(err, "Invalid request range")
 	}
 
-	// Set standard object headers.
-	setObjectHeaders(w, objInfo, hrange)
-
-	// Set any additional requested response headers.
-	setGetRespHeaders(w, r.URL.Query())
-
```

Comment:
Now that you moved this would introduce a bug in checkSourceETag() subsequently, change the function to take in 'ETag' rather than look for responseWriter to have ETag set.